include/irr/asset/IMeshPacker.h

```diff
@@ -21,13 +18,96 @@ template <typename MeshBufferType>
+    struct AllocationParams
+    {
+        size_t indexBuffSupportedSizeInBytes                    = 2147483648ull; /*  2GB*/
+        size_t vertexBuffSupportedSizeInBytes                   = 4294967296ull; /*  4GB*/
+        size_t perInstanceVertexBuffSupportedSizeInBytes        = 33554432ull;   /* 32MB*/
+        size_t MDIDataSizeInBytes                               = 2147483648ull; /*  2GB*/
+        size_t vertexBufferMinAllocSize                         = 256ull;        /* 256B*/
+        size_t indexBufferMinAllocSize                          = 512ull;        /* 512B*/
+        size_t perInstanceVertexBufferMinAllocSize              = 32ull;         /*  32B*/
+    };
+
+    template <typename MeshBufferType>
+    struct PackedMeshBuffer
+    {
+        core::smart_refctd_ptr<MeshBufferType> packedMeshBuffer = nullptr;
+    };
+
+    struct PackedMeshBufferData
+    {
+        const size_t mdiParameterByteOffset; // add to `CCPUMeshPacker::getMultiDrawIndirectBuffer()->getPointer() to get `DrawElementsIndirectCommand_t` address
+        const uint32_t mdiParameterCount;
+    };
+
+public:
+
+    IMeshPacker(const SVertexInputParams& preDefinedLayout, const AllocationParams& allocParams,uint16_t maxIndexCountPerMDIData = std::numeric_limits<uint16_t>::max())
+        :m_maxIndexCountPerMDIData(maxIndexCountPerMDIData), m_drawCount(0)
+    {
+        m_outVtxInputParams.enabledAttribFlags  = preDefinedLayout.enabledAttribFlags;
+        m_outVtxInputParams.enabledBindingFlags = preDefinedLayout.enabledBindingFlags;
+
+        memcpy(m_outVtxInputParams.attributes, preDefinedLayout.attributes, sizeof(m_outVtxInputParams.attributes));
+
+        m_outMDIData = core::make_smart_refctd_ptr<ICPUBuffer>(allocParams.MDIDataSizeInBytes);
+
+        //TODO: init allocators 
+        //m_alctrReservedSpace[0] = _IRR_ALIGNED_MALLOC(core::GeneralpurposeAddressAllocator::reserved_size(/**/), _IRR_SIMD_ALIGNMENT);
+        //m_vtxBuffAlctr = core::GeneralpurposeAddressAllocator(m_alctrReservedSpace[0], 0u, 0u, 0u, allocParams.vertexBuffSupportedSizeInBytes / allocParams.vertexbufferMinAllocSize, allocParams.vertexbufferMinAllocSize);
+
+        //m_alctrReservedSpace[1] = _IRR_ALIGNED_MALLOC(core::GeneralpurposeAddressAllocator::reserved_size(/**/), _IRR_SIMD_ALIGNMENT);
+        //m_idxBuffAlctr = core::GeneralpurposeAddressAllocator(m_alctrReservedSpace[1], 0u, 0u, 0u, allocParams.indexBuffSupportedSizeInBytes / allocParams.indexBufferMinAllocSize, allocParams.indexBufferMinAllocSize);
+
+        //m_alctrReservedSpace[2] = _IRR_ALIGNED_MALLOC(core::GeneralpurposeAddressAllocator::reserved_size(/**/), _IRR_SIMD_ALIGNMENT);
+        //m_perInsVtxBuffAlctr = core::GeneralpurposeAddressAllocator(m_alctrReservedSpace[2], 0u, 0u, 0u, allocParams.perInstanceVertexBuffSupportedSizeInBytes / allocParams.perInstanceVertexbufferMinAllocSize, allocParams.perInstanceVertexbufferMinAllocSize);
```

Comment:
don't divide the address space by the minimum allocation size, just keep it in units of `uint32_t` (so divide by `sizeof(uint32_t)`)

Reply:
see latest comment on your default parameters